lib/fs.js

```diff
@@ -1773,8 +1778,8 @@ function symlink(target, path, type_, callback_) {
  * @returns {void}
  */
 function symlinkSync(target, path, type) {
-  type = (typeof type === 'string' ? type : null);
-  if (isWindows && type === null) {
+  validateOneOf(type, 'type', ['dir', 'file', 'junction', null, undefined]);
+  if (isWindows && type == null) {
```

Comment:
This seems to change the behavior of `symlink` for Windows. Is this intentional?

Reply:
How does it change its behavior? It rejcets non-string and non-nullish values, but other than that the bahavior seems exactly the same, no?

Reply:
Yeah, it seems like the behavior should be the same, since the `undefined is one of the [possible values](https://github.com/LiviaMedeiros/node/blob/f05c86339216072559dd5c538d3140b005f76cb2/lib/fs.js#L1781) a loose equality comparison should rejects non-nullish values properly:
![image](https://github.com/nodejs/node/assets/23525618/96da0a8a-6dc4-4d5c-a527-06def1f19d15)